### PR TITLE
Rework MMFormPhotoEditor

### PR DESCRIPTION
**Pull request by tomasMizera**
image
image

I moved two photo editor-related files from `./components` to `./form/components/photo`.
There are a bunch of things that could be changed but I think it is fine for now :)

### CODE REVIEW

File: app/qml/form/editors/MMFormPhotoViewer.qml [hunk 1]
```diff
@@ -53,50 +47,58 @@ MMBaseInput {
...
+          rightMargin: __style.margin10
+          bottomMargin: __style.margin10
         }
 
         bgndColor: __style.negativeColor
         iconSource: __style.deleteIcon
         iconColor: __style.grapeColor
-        visible: root.allowEditing && root.state !== "notSet"
+
+        visible: root.editState === "enabled" && root.state !== "notSet"
```
Comment (uclaros) [inline app/qml/form/editors/MMFormPhotoViewer.qml:90]: is "notSet" a leftover? It's not in the base class but is used elsewhere in this file too

File: app/qml/form/components/photo/MMPhotoAttachment.qml [hunk 2]
```diff
@@ -42,13 +43,13 @@ Rectangle {
...
+        MMComponents.MMIcon {
           anchors.horizontalCenter: parent.horizontalCenter
           source: __style.addImageIcon
           color: __style.forestColor
         }
 
-        Text {
+        MMComponents.MMText {
           width: parent.width
           height: 24 * __dp
```
Comment (uclaros) [inline app/qml/form/components/photo/MMPhotoAttachment.qml:54]: Shouldn't the height be removed for MMTexts?

**Review by uclaros — COMMENTED** [review]
A couple of minor comments, otherwise looks good!

File: app/qml/form/editors/MMFormPhotoViewer.qml (hunk 1 above)
Comment (tomasMizera) [reply app/qml/form/editors/MMFormPhotoViewer.qml:90]: It is a part of this `MMFormPhotoViewer` file - but good catch, maybe it should not be set as a state of base class. I will add it to a sub-state group instead 👍🏻

**Comment by tomasMizera** [comment]
@uclaros your comments should be addressed now

**Review by uclaros — APPROVED** [review]

**Pull request merged**